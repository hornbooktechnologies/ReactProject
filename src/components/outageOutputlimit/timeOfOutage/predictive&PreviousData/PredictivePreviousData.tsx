import React from 'react'
import { useTranslation } from 'react-i18next'
import OutageHeader from '../../OutageHeader'
import PlannedOutageData from './PlannedOutageData'
import {TimeOfOutageModel} from '../../../../models/outageOutputLimit/TimeOfOutageModel'
import { PlannedOutageDataModel } from '../../../../models/outageOutputLimit/PlannedOutageDataModel'
import CustomPie from '../../../common/customPie/CustomPie'

const PredictivePreviousData = ({donutchartData4_1,donutchartData3_31,plannedPositive,plannedNegative,plannedNegativeStartTopresent,plannedPositiveStartTopresent}:TimeOfOutageModel) => {
  const { i18n } = useTranslation();
  const pieChartRenderSize = 160;
  return (
    <>
      <div className="uk-width-expand@m">
        <div className="uk-card mb-20">
          <OutageHeader title={i18n.t('LABELS.YEAR_START_TO_PRESENT')} />
          <div
            className="uk-grid uk-grid-small uk-child-width-1-3@s uk-child-width-1-3@m mm-10"
            uk-grid="true"
          >
            <div className="media-mb-20 pl-10">
              {plannedNegativeStartTopresent?.map((plannedObj: PlannedOutageDataModel, index: React.Key | null | undefined) => {
                return (
                  <div key={index}>
                    <PlannedOutageData
                      planTitle={i18n.t(plannedObj.planTitle)}
                      planVal={plannedObj.planVal}
                      planpreviousVal={plannedObj.planpreviousVal}
                      suffix={plannedObj.suffix}
                      subject={plannedObj.subject}
                      className={plannedObj.className}
                      iconClassName={plannedObj.iconClassName}
                      variation={plannedObj.variation}
                    />
                  </div>
                )
              })}
            </div>
            <div className="pl-10">
              {plannedPositiveStartTopresent?.map((plannedObj:PlannedOutageDataModel, index: React.Key | null | undefined) => {
                return (
                  <div key={index}>
                    <PlannedOutageData
                      planTitle={i18n.t(plannedObj.planTitle)}
                      planVal={plannedObj.planVal}
                      planpreviousVal={plannedObj.planpreviousVal}
                      suffix={plannedObj.suffix}
                      subject={plannedObj.subject}
                      className={plannedObj.className}
                      iconClassName={plannedObj.iconClassName}
                      variation={plannedObj.variation}
                    />
                  </div>
                )
              })}
            </div>
            <div className="uk-text-center mt--35">
              <div className="uk-flex uk-flex-center uk-flex-middle uk-flex-column chart-100 uk-margin-auto-left uk-margin-auto-right">
                {donutchartData4_1 !== undefined &&
                  donutchartData4_1 !== null && (
                  <CustomPie 
                     data={donutchartData4_1} 
                     width={pieChartRenderSize}
                     height={pieChartRenderSize}
                     />
                  )}
              </div>
            </div>
          </div>
        </div>

        <div className="uk-card">
          <OutageHeader title={i18n.t('LABELS.PRESENT_TO_YEAR_END')} />
          <div
            className="uk-grid uk-grid-small uk-child-width-1-3@s uk-child-width-1-3@m mm-10"
            uk-grid="true"
          >
            <div className="media-mb-20 pl-10">
              {plannedNegative?.map((plannedObj:PlannedOutageDataModel, index: React.Key | null | undefined) => {
                 return (
                  <div key={index}>
                    <PlannedOutageData
                      planTitle={i18n.t(plannedObj.planTitle)}
                      planVal={plannedObj.planVal}
                      planpreviousVal={plannedObj.planpreviousVal}
                      suffix={plannedObj.suffix}
                      subject={plannedObj.subject}
                      className={plannedObj.className}
                      iconClassName={plannedObj.iconClassName}
                      variation={plannedObj.variation}
                    />
                  </div>
                )
              })}
            </div>
            <div className="pl-10">
              {plannedPositive?.map((plannedObj:PlannedOutageDataModel, index: React.Key | null | undefined) => {
                return (
                  <div key={index}>
                    <PlannedOutageData
                      planTitle={i18n.t(plannedObj.planTitle)}
                      planVal={plannedObj.planVal}
                      planpreviousVal={plannedObj.planpreviousVal}
                      suffix={plannedObj.suffix}
                      subject={plannedObj.subject}
                      className={plannedObj.className}
                      iconClassName={plannedObj.iconClassName}
                      variation={plannedObj.variation}
                    />
                  </div>
                )
              })}
            </div>

            <div className="uk-text-center mt--35">
              <div className="uk-flex uk-flex-center uk-flex-middle uk-flex-column chart-100 uk-margin-auto-left uk-margin-auto-right">
                {donutchartData3_31 !== undefined &&
                  donutchartData3_31 !== null && (
                  <CustomPie 
                     data={donutchartData3_31} 
                     width={pieChartRenderSize}
                     height={pieChartRenderSize}
                     />
                  )}
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  )
}

export default PredictivePreviousData;
