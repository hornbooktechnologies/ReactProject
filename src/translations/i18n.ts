import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
