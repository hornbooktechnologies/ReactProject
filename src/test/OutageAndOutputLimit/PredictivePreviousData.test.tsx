import React from "react";
import renderer from "react-test-renderer";
import PredictivePreviousData from "../../components/outageOutputlimit/timeOfOutage/predictive&PreviousData/PredictivePreviousData";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";

describe("Component: PredictivePreviousData", () => {
  it("PredictivePreviousData renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <PredictivePreviousData/>
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
