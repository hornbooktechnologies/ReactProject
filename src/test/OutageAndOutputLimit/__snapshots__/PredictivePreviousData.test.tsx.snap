// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PredictivePreviousData PredictivePreviousData renders correctly 1`] = `
<div
  className="uk-width-expand@m"
>
  <div
    className="uk-card mb-20"
  >
    <h5
      className="mb-24 font-18"
    >
      LABELS.YEAR_START_TO_PRESENT
    </h5>
    <div
      className="uk-grid uk-grid-small uk-child-width-1-3@s uk-child-width-1-3@m mm-10"
      uk-grid="true"
    >
      <div
        className="media-mb-20 pl-10"
      />
      <div
        className="pl-10"
      />
      <div
        className="uk-text-center mt--35"
      >
        <div
          className="uk-flex uk-flex-center uk-flex-middle uk-flex-column chart-100 uk-margin-auto-left uk-margin-auto-right"
        />
      </div>
    </div>
  </div>
  <div
    className="uk-card"
  >
    <h5
      className="mb-24 font-18"
    >
      LABELS.PRESENT_TO_YEAR_END
    </h5>
    <div
      className="uk-grid uk-grid-small uk-child-width-1-3@s uk-child-width-1-3@m mm-10"
      uk-grid="true"
    >
      <div
        className="media-mb-20 pl-10"
      />
      <div
        className="pl-10"
      />
      <div
        className="uk-text-center mt--35"
      >
        <div
          className="uk-flex uk-flex-center uk-flex-middle uk-flex-column chart-100 uk-margin-auto-left uk-margin-auto-right"
        />
      </div>
    </div>
  </div>
</div>
`;
